caffe2/serialize/inline_container.h

```diff
@@ -203,11 +203,21 @@ class TORCH_API PyTorchStreamReader final {
   size_t additional_reader_size_threshold_;
 };
 
+namespace {
+
+size_t default_seek_func(size_t nbytes) {
+  TORCH_CHECK(false, "attempting to write record metadata but seek_func unimplemented, please implement seek_func");
+  return 0;
+}
+
+} // namespace
+
 class TORCH_API PyTorchStreamWriter final {
  public:
   explicit PyTorchStreamWriter(const std::string& archive_name);
   explicit PyTorchStreamWriter(
-      const std::function<size_t(const void*, size_t)> writer_func);
+      const std::function<size_t(const void*, size_t)> writer_func,
+      const std::function<size_t(size_t)> seek_func = default_seek_func);
```

Comment:
```suggestion
      const std::function<size_t(size_t)> seek_func = {});
```
If you don't want to deal with the default, std::function has an empty state that works very much like an optional and you can do `if (seek_func)` on it.